.github/workflows/functional-baremetal.yaml

```
@@ -2,6 +2,13 @@ name: functional-baremetal
 on:
   pull_request:
     paths:
+      - 'openstack/auth_env.go'
+      - 'openstack/client.go'
+      - 'openstack/endpoint.go'
+      - 'openstack/endpoint_location.go'
+      - 'openstack/config/provider_client.go'
+      - 'openstack/utils/choose_version.go'
+      - 'openstack/utils/discovery.go'
```

Isn't it a bit fragile to list all files individually? Should we consider using a regex? Also, how about files in `openstack/common` or `openstack/utils/base_endpoint.go`?

What do you think of the following list?
```suggestion
      - 'openstack/*.go'
      - 'openstack/common/**'
      - 'openstack/config/**'
      - 'openstack/utils/**'
```

---

It is fragile, but I had intentionally avoided doing that because I wanted to avoid running functional tests on changes to `*_test.go` files. What do you think?

---

I'm of the opinion that more CI is better than not enough. There aren't too many test files, and the probability we change them without also changing the code it tests is pretty low:
```
❯ ls -1 ./openstack/*_test.go ./openstack/(common|config|utils)/**/*_test.go 
./openstack/client_test.go
./openstack/common/extensions/testing/requests_test.go
./openstack/config/clouds/clouds_test.go
./openstack/utils/testing/base_endpoint_test.go
./openstack/utils/testing/choose_version_test.go
```

However, I'm not going to block on this.